src/main/java/org/terasology/launcher/game/GameStarter.java

```diff
@@ -97,23 +96,27 @@ public Process call() throws IOException {
     /**
      * @return the executable {@code java} file to run the game with
      */
-    Path getRuntimePath() {
+    Path getRuntimePath(Semver engineVersion) throws GameVersionNotSupportedException {
+        if (VersionHistory.JAVA17.isProvidedBy(engineVersion)) {
+            // throw exception as the version is not supported
```

Comment:
I'm wondering whether it would make sense to log a bit more here, why this is not supported to allow deducing this directly from the logs instead of having to look at the comment in `VersionHistory.java`.

Reply:
I think this would require some (minor) rework of the `VersionHistory` to carry that information in the type. 

However, I thought that this case should usually not appear, and if it appears, the version number of the unsupported release should be enough hint for us to figure out what is wrong. A user might not gain much from knowing in which commit ID something was changed 🤷